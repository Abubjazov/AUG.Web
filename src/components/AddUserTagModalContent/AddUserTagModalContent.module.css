.root {
  display: flex;
  flex-direction: column;
  gap: 20px;
  width: 80vw;
  color: var(--black);
  animation: var(--animation-visible);
}

.title {
  font-size: 48px;
  font-weight: 900;
  line-height: 48px;
}

.text {
  font-size: 16px;
  font-weight: 500;
  line-height: 16px;
}

.buttons {
  display: flex;
  gap: 20px;
}

.button {
  height: 25px;
  margin-top: 20px;
}
