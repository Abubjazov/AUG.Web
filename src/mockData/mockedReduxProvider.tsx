import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import {
  mockCommunityTags,
  mockDapplets,
  mockMyDapplets,
  mockMyTags,
} from './mockData'
import dappletsReducer from '../store/slices/dappletsSlice'
import layoutReducer from '../store/slices/layoutSlice'
import myDappletsReducer from '../store/slices/myDappletsSlice'

export const defaultMockState = {
  layout: {
    menuOpened: false,
    dappletSettingsOpened: false,
  },

  myDapplets: {
    myDapplets: [...mockMyDapplets],
    myTags: [...mockMyTags],
  },

  dapplets: {
    dapplets: [...mockDapplets],
    tags: [...mockCommunityTags],
  },
}

export const mockedReduxProvider = ({
  children,
}: PropsWithChildren<object>) => {
  const store = configureStore({
    reducer: {
      layout: layoutReducer,
      dapplets: dappletsReducer,
      myDapplets: myDappletsReducer,
    },
    preloadedState: {
      ...defaultMockState,
    },
  })

  return <Provider store={store}>{children}</Provider>
}
