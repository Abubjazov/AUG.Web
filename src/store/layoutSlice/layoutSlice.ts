import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'

import type { RootState } from '../index'

type TLayout = { menuOpened: boolean; dappletSettingsOpened: boolean }

const windowInnerWidth = window.innerWidth

export const getInitialState = (windowInnerWidth: number, setPoint: number) =>
  windowInnerWidth > setPoint ? true : false

const initialState: TLayout = {
  menuOpened: getInitialState(windowInnerWidth, 1300),
  dappletSettingsOpened: getInitialState(windowInnerWidth, 1600),
}

export const layoutSlice = createSlice({
  name: 'layout',
  initialState,
  reducers: {
    setDappletSettingsState: (state, action: PayloadAction<boolean>) => {
      state.dappletSettingsOpened = action.payload
    },

    setMenuState: (state, action: PayloadAction<boolean>) => {
      state.menuOpened = action.payload
    },
  },
})

export const { setDappletSettingsState, setMenuState } = layoutSlice.actions

export const selectLayout = (state: RootState) => state.layout

export default layoutSlice.reducer
