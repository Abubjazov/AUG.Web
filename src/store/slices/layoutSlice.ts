import { ReactNode } from 'react'

import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'
import { getInitialState } from 'utils/getInItialState/getInItialState'

import type { RootState } from '../index'

type TLayout = {
  menuOpened: boolean
  dappletSettingsOpened: boolean
  modalState: boolean
  modalContent: ReactNode
}

const windowInnerWidth = window.innerWidth

const initialState: TLayout = {
  menuOpened: getInitialState(windowInnerWidth, 1300),
  dappletSettingsOpened: getInitialState(windowInnerWidth, 1600),
  modalState: true,
  modalContent: null,
}

export const layoutSlice = createSlice({
  name: 'layout',
  initialState,
  reducers: {
    setDappletSettingsState: (state, action: PayloadAction<boolean>) => {
      state.dappletSettingsOpened = action.payload
    },

    setMenuState: (state, action: PayloadAction<boolean>) => {
      state.menuOpened = action.payload
    },

    setModalState: (state, action: PayloadAction<boolean>) => {
      state.modalState = action.payload
    },

    setModalContent: (state, action: PayloadAction<ReactNode>) => {
      state.modalContent = action.payload
    },
  },
})

export const {
  setDappletSettingsState,
  setMenuState,
  setModalState,
  setModalContent,
} = layoutSlice.actions

export const selectLayout = (state: RootState) => state.layout

export default layoutSlice.reducer
