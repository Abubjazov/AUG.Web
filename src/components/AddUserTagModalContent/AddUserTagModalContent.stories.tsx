import type { Meta, StoryObj } from '@storybook/react'
import MockedProvider from 'mockData/mockedReduxProvider'

import AddUserTagModalContent, {
  AddUserTagModalContentProps,
} from './AddUserTagModalContent'

import '/src/index.css'

const meta: Meta<AddUserTagModalContentProps> = {
  component: AddUserTagModalContent,
  title: 'COMPONENTS/Add User Tag Modal Content',
  parameters: {
    docs: {
      description: {
        component: 'Interface component - AddUserTagModalContent',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    dappletId: {
      description: 'Current dapplet ID',
    },
  },
}

export default meta

type Story = StoryObj<AddUserTagModalContentProps>

export const Default: Story = {
  args: {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
  },

  decorators: [
    Story => (
      <MockedProvider>
        <Story />
      </MockedProvider>
    ),
  ],
}
