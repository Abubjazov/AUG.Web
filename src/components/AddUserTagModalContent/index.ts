import AddUserTagModalContent from './AddUserTagModalContent'

export default AddUserTagModalContent
