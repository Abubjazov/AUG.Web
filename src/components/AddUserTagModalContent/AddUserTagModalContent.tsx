import { FC } from 'react'

import TagsGroup from 'components/TagsGroup'
import { useAppDispatch, useAppSelector } from 'store/hooks'
import { setModalState } from 'store/slices/layoutSlice'
import BaseButton from 'uikit/BaseButton'
import { BaseButtonMode } from 'uikit/BaseButton/BaseButton'
import { SmartTagMode } from 'uikit/SmartTag/SmartTag'

import styles from './AddUserTagModalContent.module.css'

export enum ERenderMode {
  WELCOME = 'welcome',
  LOGIN = 'login',
  REGISTRATION = 'registration',
}

export interface AddUserTagModalContentProps {
  dappletId: string
}

const AddUserTagModalContent: FC<AddUserTagModalContentProps> = ({
  dappletId,
}) => {
  const dispatch = useAppDispatch()

  const { userTags, tagOperationGoing } = useAppSelector(
    state => state.userData,
  )

  return (
    <div className={styles['root']}>
      <TagsGroup
        menuOpened={true}
        tags={userTags}
        title={'My tags'}
        tagMode={SmartTagMode.MY_TAG_MODAL}
        titleUppercase
        tagOperationGoing={tagOperationGoing}
        dappletId={dappletId}
      />

      <div className={styles.buttons}>
        <BaseButton
          userStyles={styles.button}
          label={'Close'}
          mode={BaseButtonMode.CONTAINED_RED}
          onClick={() => dispatch(setModalState(false))}
        />
      </div>
    </div>
  )
}

export default AddUserTagModalContent
