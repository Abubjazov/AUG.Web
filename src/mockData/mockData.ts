import { IDapplet, ITag } from 'store/slices/dappletsSlice'
import { IList, IUserDapplet } from 'store/slices/userDataSlice'

export const mockUserTags: ITag[] = [
  { tagId: 'darP5Jyz8yirTMsfY9RP', tagName: 'Twitter' },
  { tagId: 'JW3UFtZ5HgATcwldsJ1T', tagName: 'Facebook' },
  { tagId: '6UKxNrMzte6RCWIlssvM', tagName: 'Top 10' },
  { tagId: 'ZbEuko9nOlQYKG6I1Efx', tagName: 'Test' },
  { tagId: '521keF1Adymn8Oo896vv', tagName: 'ToDo' },
]

export const mockUserLists: IList[] = [
  { listId: 'XfLdXIykGHsgIjdJkTaE', listName: 'My special list' },
  { listId: 'c6rClgOltChco2XnyVDp', listName: 'Financial list' },
]

export const mockUserDapplets: IUserDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    userTags: ['JW3UFtZ5HgATcwldsJ1T', '521keF1Adymn8Oo896vv'],
    dappletState: true,
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    userTags: [],
    dappletState: true,
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    userTags: 'N/A',
    dappletState: true,
  },
]

export const mockCommunityTags: ITag[] = [
  { tagId: 'XoEVTHXhfMAjIWsi4wxY', tagName: 'Top 100' },
  { tagId: '3LYUFQTbdNULwBgCkfXx', tagName: 'Finances' },
  { tagId: 'XZ2zPZqa4fPxM9mLA9xR', tagName: 'Media' },
  { tagId: 'C69B5fQ7ccW6Q1SC0yow', tagName: 'Favourites' },
]

export const mockDapplets: IDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    appOwner: 'Satoshi Nakamoto',
    circulatingSupply: '19534212',
    communityTags: ['3LYUFQTbdNULwBgCkfXx', 'C69B5fQ7ccW6Q1SC0yow'],
    date: '1230757200',
    fullDesc:
      'Bitcoin is a peer-to-peer online currency, meaning that all transactions happen directly between equal, independent network participants, without the need for any intermediary to permit or facilitate them. Bitcoin was created, according to Nakamoto’s own words, to allow “online payments to be sent directly from one party to another without going through a financial institution.”',
    fullyDilutedMarketCap: '738866583732',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-1.01',
    maxSupply: '21000000',
    name: 'Bitcoin Contracts',
    shortDesc:
      'Bitcoin’s original inventor is known under a pseudonym, Satoshi Nakamoto. ',
    shortName: 'BTC',
    totalSupply: '19534106',
    volume: '10.9',
    volumePerMarketCap: '1.58',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHGCOPY',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    appOwner: 'XRP Ledger',
    circulatingSupply: '53615837759',
    communityTags: 'N/A',
    date: '1620939600',
    fullDesc:
      'Launched in 2021, the XRP Ledger (XRPL) is an open-source, permissionless and decentralized technology. Benefits of the XRP Ledger include its low-cost ($0.0002 to transact), speed (settling transactions in 3-5 seconds), scalability (1,500 transactions per second) and inherently green attributes (carbon-neutral and energy-efficient). The XRP Ledger also features the first decentralized exchange (DEX) and custom tokenization capabilities built into the protocol. Since 2012, the XRP Ledger has been operating reliably, having closed 70 million ledgers.',
    fullyDilutedMarketCap: '65183420834',
    logo: '/public/images/notAvailable.svg',
    marketCap: '5.66',
    maxSupply: '100000000000',
    name: 'XRP Ledger',
    shortDesc:
      'Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.',
    shortName: 'XRP',
    totalSupply: '99988316618',
    volume: '132.26',
    volumePerMarketCap: '5.37',
  },
  {
    dappletId: 'QbWG3sKvfgFcP5RtskMp',
    appOwner: 'Realcoin',
    circulatingSupply: '85367759449',
    communityTags: [],
    date: '1388523600',
    fullDesc:
      'Launched in 2014, Tether is a blockchain-enabled platform designed to facilitate the use of fiat currencies in a digital manner. Tether works to disrupt the conventional financial system via a more modern approach to money. Tether has made headway by giving customers the ability to transact with traditional currencies across the blockchain, without the inherent volatility and complexity typically associated with a digital currency. As the first blockchain-enabled platform to facilitate the digital use of traditional currencies (a familiar, stable accounting unit), Tether has democratised cross-border transactions across the blockchain.',
    fullyDilutedMarketCap: '88708606718',
    logo: '/dappletsLogos/tether.png',
    marketCap: '0.01',
    maxSupply: 'N/A',
    name: 'Tether USDt',
    shortDesc:
      'Tether works to disrupt the conventional financial system via a more modern approach to money.',
    shortName: 'USDT',
    totalSupply: '88623656724',
    volume: '41.44',
    volumePerMarketCap: '32.16',
  },
]
