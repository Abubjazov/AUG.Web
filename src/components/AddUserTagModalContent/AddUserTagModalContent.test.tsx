import { fireEvent, render, screen } from '@testing-library/react'
import { mockedReduxProvider as MProvider } from 'mockData/mockedReduxProvider'
import * as actions from 'store/slices/layoutSlice'

import AddUserTagModalContent from './AddUserTagModalContent'

describe('AddUserTagModalContent', () => {
  test('should render AddUserTagModalContent default', () => {
    const { asFragment } = render(
      <MProvider>
        <AddUserTagModalContent dappletId={'ECNk2nNngwGXouvMpjWt'} />
      </MProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should call setModalState function on Close button click', () => {
    const mockedSetModalState = vi.spyOn(actions, 'setModalState')

    render(
      <MProvider>
        <AddUserTagModalContent dappletId={'ECNk2nNngwGXouvMpjWt'} />
      </MProvider>,
    )

    fireEvent.click(screen.getByText('Close'))

    expect(mockedSetModalState).toHaveBeenCalledTimes(1)
    expect(mockedSetModalState).toHaveBeenCalledWith(false)
  })
})
