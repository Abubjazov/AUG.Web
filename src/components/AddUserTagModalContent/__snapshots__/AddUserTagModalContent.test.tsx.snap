// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddUserTagModalContent > should render AddUserTagModalContent default 1`] = `
<DocumentFragment>
  <div
    class="_root_24aef0"
  >
    <div
      class="_root_435026  "
    >
      <span
        class="_title_435026 _uppercase_435026"
      >
        My tags
      </span>
      <div
        class="_list_435026"
      >
        <div
          class="_root_849b90 _my-tag-modal_849b90 _list-item_435026"
          data-testid="smart-tag"
          draggable="true"
        >
          <span>
            Twitter
          </span>
        </div>
        <div
          class="_root_849b90 _my-tag-modal_849b90 _list-item_435026"
          data-testid="smart-tag"
          draggable="true"
        >
          <span>
            Facebook
          </span>
        </div>
        <div
          class="_root_849b90 _my-tag-modal_849b90 _list-item_435026"
          data-testid="smart-tag"
          draggable="true"
        >
          <span>
            Top 10
          </span>
        </div>
        <div
          class="_root_849b90 _my-tag-modal_849b90 _list-item_435026"
          data-testid="smart-tag"
          draggable="true"
        >
          <span>
            Test
          </span>
        </div>
        <div
          class="_root_849b90 _my-tag-modal_849b90 _list-item_435026"
          data-testid="smart-tag"
          draggable="true"
        >
          <span>
            ToDo
          </span>
        </div>
      </div>
    </div>
    <div
      class="_buttons_24aef0"
    >
      <button
        class="_root_91d6ec _contained-red_91d6ec  _button_24aef0"
        data-testid="base-button"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
